### PR TITLE
Replace Dotnet.ProjInf with Ionide.ProjInfo.

### PR DESCRIPTION
**Pull request by nojaf**
As mentioned in https://github.com/dotnet/fsharp/discussions/13646.
I would later like to use this in my test over at https://github.com/safesparrow/fsharp/pull/1.

### CODE REVIEW

**Comment by nojaf** [comment]
@vzarytovskii looks like I'm still missing a dependency

```
tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj(0,0): error NU1101: (NETCORE_ENGINEERING_TELEMETRY=Restore) Unable to find package SemanticVersioning. No packages exist with this id in source(s): dotnet-eng, dotnet-public, dotnet-tools, dotnet5, dotnet5-transport, vs-impl, vssdk
```

File: tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj [hunk 1]
```diff
@@ -167,7 +168,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Dotnet.ProjInfo" Version="0.37.0" />
+    <PackageReference Include="Microsoft.Build.Framework" Version="17.2.0" Condition="'$(TargetFramework)' == 'net6.0'" ExcludeAssets="runtime" PrivateAssets="all" />
```
Comment (vzarytovskii) [inline tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj:171]: We should use versions from `eng\Versions.props`

File: tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
   <ItemGroup>
     <Compile Include="..\fsharp\TestHelpers.fs" Link="TestHelpers.fs" />
+    <Compile Include="ProjectInfoHelpers.fs" Condition="'$(TargetFramework)' == 'net6.0'" />
```
Comment (vzarytovskii) [inline tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj:24]: Why is target framework condition is necessary?

**Comment by vzarytovskii** [comment]
> @vzarytovskii looks like I'm still missing a dependency
> 
> ```
> tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj(0,0): error NU1101: (NETCORE_ENGINEERING_TELEMETRY=Restore) Unable to find package SemanticVersioning. No packages exist with this id in source(s): dotnet-eng, dotnet-public, dotnet-tools, dotnet5, dotnet5-transport, vs-impl, vssdk
> ```

Is it the only one missing? Will the last version work, or some specific one is required?

File: tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj (hunk 2 above)
Comment (nojaf) [reply tests/FSharp.Compiler.Service.Tests/FSharp.Compiler.Service.Tests.fsproj:24]: `Ionide.ProjInfo` is a `.net6.0` only thing, https://www.nuget.org/packages/Ionide.ProjInfo#supportedframeworks-body-tab

**Comment by nojaf** [comment]
> Is it the only one missing? Will the last version work, or some specific one is required?

Yeah, the latest should work.

**Comment by nojaf** [comment]
@vzarytovskii last week I was finally able to deduce that the current approach is never going to work.

The project `FSharp.Compiler.Service` contains direct dependencies to

```xml
    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildVersion)" />
    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildVersion)" />
    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildVersion)" />
```

which conflicts with the dynamic loading of the correct MSBuild assemblies that `Ionide.ProjInfo` is trying to apply. There are some unit tests depending on this, so we cannot alter `FSharp.Compiler.Service.Tests`.

Would you be ok with having an additional unit test project where we can exclude these references at runtime in order for `ProjInfo` to work as expected?

**Comment by vzarytovskii** [comment]
> @vzarytovskii last week I was finally able to deduce that the current approach is never going to work.
> 
> The project `FSharp.Compiler.Service` contains direct dependencies to
> 
> ```xml
>     <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildVersion)" />
>     <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildVersion)" />
>     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildVersion)" />
> ```
> 
> which conflicts with the dynamic loading of the correct MSBuild assemblies that `Ionide.ProjInfo` is trying to apply. There are some unit tests depending on this, so we cannot alter `FSharp.Compiler.Service.Tests`.
> 
> Would you be ok with having an additional unit test project where we can exclude these references at runtime in order for `ProjInfo` to work as expected?

I guess it'll be ok, as long as we constraint it to have specific type of tests.

**Comment by nojaf** [comment]
@vzarytovskii am I still missing anything for this new test project?

**Pull request closed without merging**